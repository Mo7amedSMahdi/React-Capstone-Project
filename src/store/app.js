import { createSlice } from '@reduxjs/toolkit';
import moment from 'moment';
import { apiCallBegan } from './api';
import countries from '../data/coutriesLatLong';

const appId = '47ae443f230aa695cca70832bae5b260';
const slice = createSlice({
  name: 'airPollution',
  initialState: {
    list: [],
    loading: false,
    lastFetch: null,
  },
  reducers: {
    dataRequested: (state) => {
      state.loading = true;
    },
    dataRequestFailed: (state) => {
      state.loading = false;
    },
    dataReceived: (state, action) => {
      state.list = action.payload;
      state.loading = false;
      state.lastFetch = Date.now();
    },
  },
});

export const { dataRequested, dataRequestFailed, dataReceived } = slice.actions;

export default slice.reducer;

export const getData = () => async (dispatch, getState) => {
  const { lastFetch } = getState().app;

  const diffInMinutes = moment().diff(moment(lastFetch), 'minutes');
  if (diffInMinutes < 10) return;

  dispatch(
    apiCallBegan({
      countries,
      appId,
      onStart: dataRequested.type,
      onSuccess: dataReceived.type,
      onError: dataRequestFailed.type,
    }),
  );
};
