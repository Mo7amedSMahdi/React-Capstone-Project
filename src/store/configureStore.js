import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';
import api from './middleware/api';
import appInfo from './app';

const rootReducer = combineReducers({
  app: appInfo,
});

const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(api),
});

export default store;
