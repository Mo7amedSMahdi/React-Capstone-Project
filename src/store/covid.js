import { createSlice } from '@reduxjs/toolkit';

const baseURL = 'https://api.covid19tracking.narrativa.com/api/';
const inialState = [];

const slice = createSlice({
  name: 'covid',
  inialState: inialState,
});
