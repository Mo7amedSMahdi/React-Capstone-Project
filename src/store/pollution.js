import { createSlice } from '@reduxjs/toolkit';
import moment from 'moment';
import { apiCallBegan } from './api';
import countries from '../data/coutriesLatLong';

const appId = '47ae443f230aa695cca70832bae5b260';
const slice = createSlice({
  name: 'airPollution',
  initialState: {
    list: [],
    loading: false,
    lastFetch: null,
  },
  reducers: {
    dataRequested: (state) => {
      state.loading = true;
    },
    dataRequestFailed: (state) => {
      state.loading = false;
    },
    dataReceived: (state, action) => {
      state.list = action.payload.map((data, index) => {
        if (data.coord.lat === countries[index].latitude) {
          switch (data.list[0].main.aqi) {
            case 1:
              data.list[0].main.aqi = 'Good';
              break;
            case 2:
              data.list[0].main.aqi = 'Fair';
              break;
            case 3:
              data.list[0].main.aqi = 'Moderate';
              break;
            case 4:
              data.list[0].main.aqi = 'Poor';
              break;
            case 5:
              data.list[0].main.aqi = 'Very Poor';
              break;
            default:
              break;
          }
          return { ...data, country: countries[index].country, map: countries[index].map };
        }
      });
      state.loading = false;
      state.lastFetch = Date.now();
    },
    getCountry: (state, action) => {
      state.list.filter((p) => p.country === action.payload);
    },
  },
});

export const { dataRequested, dataRequestFailed, dataReceived, getCountry } = slice.actions;

export default slice.reducer;

export const getData = () => async (dispatch, getState) => {
  const { lastFetch } = getState().pollution;

  const diffInMinutes = moment().diff(moment(lastFetch), 'minutes');
  if (diffInMinutes < 10) return;

  dispatch(
    apiCallBegan({
      countries,
      appId,
      onStart: dataRequested.type,
      onSuccess: dataReceived.type,
      onError: dataRequestFailed.type,
    }),
  );
};
