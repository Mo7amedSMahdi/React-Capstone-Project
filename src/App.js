import React from 'react';
import { Provider } from 'react-redux';
import store from './store/configureStore';
import Home from './components/home/Home';

function App() {
  return (
    <Provider store={store}>
      <Home />
    </Provider>
  );
}

export default App;
