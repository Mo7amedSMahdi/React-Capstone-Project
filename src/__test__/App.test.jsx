import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../store/configureStore';
import App from '../App';

describe('App component', () => {
  test('renders', () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
