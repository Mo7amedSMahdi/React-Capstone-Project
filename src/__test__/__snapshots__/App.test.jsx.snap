// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header flex"
      >
        <div
          class="header-title flex"
        >
          <h1>
            Eu Air Pollution
          </h1>
        </div>
        <ul
          class="nav nav--buttons"
        >
          <li>
            <button
              class="btn"
              type="button"
            >
              <i
                class="ri-mic-line"
              />
            </button>
          </li>
          <li>
            <button
              class="btn"
              type="button"
            >
              <i
                class="ri-settings-2-line"
              />
            </button>
          </li>
        </ul>
      </header>
      <main>
        <section
          class="main-section"
        >
          <div
            class="gooey"
          >
            <div
              class="gooey-container"
            >
              <span
                class="dot"
              />
              <div
                class="dots"
              >
                <span />
                <span />
                <span />
              </div>
            </div>
          </div>
          <div
            class="main-header flex flex--column"
          >
            <div
              class="header-intro flex"
            >
              <img
                alt="Europe map mask"
                src="eu-map.svg"
              />
              <div
                class="intro-title flex flex--column"
              >
                <h2>
                  Europe
                </h2>
                <p>
                  Population: 746.4 million
                </p>
              </div>
            </div>
          </div>
          <div
            class="countries"
          >
            <div
              class="header-search"
            >
              <input
                class="input-search"
                placeholder="Search by country name"
                type="search"
                value=""
              />
            </div>
            <div
              class="countries-cards grid"
            >
              <p>
                Loading
              </p>
            </div>
          </div>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <header
      class="header flex"
    >
      <div
        class="header-title flex"
      >
        <h1>
          Eu Air Pollution
        </h1>
      </div>
      <ul
        class="nav nav--buttons"
      >
        <li>
          <button
            class="btn"
            type="button"
          >
            <i
              class="ri-mic-line"
            />
          </button>
        </li>
        <li>
          <button
            class="btn"
            type="button"
          >
            <i
              class="ri-settings-2-line"
            />
          </button>
        </li>
      </ul>
    </header>
    <main>
      <section
        class="main-section"
      >
        <div
          class="gooey"
        >
          <div
            class="gooey-container"
          >
            <span
              class="dot"
            />
            <div
              class="dots"
            >
              <span />
              <span />
              <span />
            </div>
          </div>
        </div>
        <div
          class="main-header flex flex--column"
        >
          <div
            class="header-intro flex"
          >
            <img
              alt="Europe map mask"
              src="eu-map.svg"
            />
            <div
              class="intro-title flex flex--column"
            >
              <h2>
                Europe
              </h2>
              <p>
                Population: 746.4 million
              </p>
            </div>
          </div>
        </div>
        <div
          class="countries"
        >
          <div
            class="header-search"
          >
            <input
              class="input-search"
              placeholder="Search by country name"
              type="search"
              value=""
            />
          </div>
          <div
            class="countries-cards grid"
          >
            <p>
              Loading
            </p>
          </div>
        </div>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
